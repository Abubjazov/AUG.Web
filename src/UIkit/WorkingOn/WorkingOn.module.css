.root {
  text-align: left;
  transition: all var(--transition-delay) linear;
  animation: visible 0.5s ease-in-out;
}

.ds-closed {
  display: none;
}

.title {
  font-size: 24px;
  font-weight: 600;
  line-height: 24px;
  text-transform: uppercase;
}

.list {
  display: flex;
  flex-direction: column;
}

.list-item {
  display: flex;
  align-items: center;
  margin-top: 10px;
}

.list-item-text {
  margin-left: 5px;
  font-size: 16px;
  font-weight: 400;
  line-height: 16px;
  color: var(--light-textgray);
}
