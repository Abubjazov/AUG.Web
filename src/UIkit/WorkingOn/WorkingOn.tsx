import { FC } from 'react'

import SvgIcon from 'uikit/SvgIcon'
import { combineClasses as cc } from 'utils/combineClasses/combineClasses'

import styles from './WorkingOn.module.css'

export interface WorkingOnProps {
  userStyles?: string
  dsOpened: boolean
}

const WorkingOn: FC<WorkingOnProps> = ({ dsOpened, userStyles = '' }) => {
  return (
    <div
      className={cc([
        styles.root,
        dsOpened ? '' : styles['ds-closed'],
        userStyles,
      ])}
    >
      <span className={styles['title']}>Working on</span>

      <div className={styles['list']}>
        <div className={styles['list-item']}>
          <SvgIcon icon={'redcross'} />
          <span className={styles['list-item-text']}>twitter.com</span>
        </div>

        <div className={styles['list-item']}>
          <SvgIcon icon={'redcross'} />
          <span className={styles['list-item-text']}>
            twitter.com/randomuser
          </span>
        </div>

        <div className={styles['list-item']}>
          <SvgIcon icon={'redcross'} />
          <span className={styles['list-item-text']}>facebook.com</span>
        </div>

        <div className={styles['list-item']}>
          <SvgIcon icon={'redcross'} />
          <span className={styles['list-item-text']}>
            facebook.com/randomuser
          </span>
        </div>
      </div>
    </div>
  )
}

export default WorkingOn
