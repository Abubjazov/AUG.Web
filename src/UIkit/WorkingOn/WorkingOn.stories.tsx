import type { Meta, StoryObj } from '@storybook/react'

import WorkingOn, { WorkingOnProps } from './WorkingOn'

import '/src/index.css'

const meta: Meta<WorkingOnProps> = {
  component: WorkingOn,
  title: 'UIkit/Working On',
  parameters: {
    docs: {
      description: {
        component: 'Stub for the WorkingOn component',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    dsOpened: { description: 'Hiding the list' },
    userStyles: {
      description: 'Passing additional custom styles for root element',
    },
  },
}

export default meta

type Story = StoryObj<WorkingOnProps>

export const Default: Story = {
  args: {
    dsOpened: true,
  },
}
