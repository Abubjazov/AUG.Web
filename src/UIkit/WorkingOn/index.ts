import WorkingOn from './WorkingOn'

export default WorkingOn
