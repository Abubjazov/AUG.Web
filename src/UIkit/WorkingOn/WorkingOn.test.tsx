import { render } from '@testing-library/react'

import WorkingOn from './WorkingOn'

describe('WorkingOn', () => {
  test('should render WorkingOn default', () => {
    const { asFragment } = render(<WorkingOn dsOpened={true} />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render WorkingOn when the dapplet settings is closed', () => {
    const { asFragment } = render(<WorkingOn dsOpened={false} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
