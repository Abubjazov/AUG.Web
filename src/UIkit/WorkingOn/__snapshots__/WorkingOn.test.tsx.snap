// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WorkingOn > should render WorkingOn default 1`] = `
<DocumentFragment>
  <div
    class="_root_fcc54c  "
  >
    <span
      class="_title_fcc54c"
    >
      Working on
    </span>
    <div
      class="_list_fcc54c"
    >
      <div
        class="_list-item_fcc54c"
      >
        <div
          class="_circle_fcc54c"
        />
        <a
          class="_list-item-text_fcc54c"
          href="https://telegram.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          telegram.org
        </a>
      </div>
      <div
        class="_list-item_fcc54c"
      >
        <div
          class="_circle_fcc54c"
        />
        <a
          class="_list-item-text_fcc54c"
          href="https://www.whatsapp.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          whatsapp.com
        </a>
      </div>
      <div
        class="_list-item_fcc54c"
      >
        <div
          class="_circle_fcc54c"
        />
        <a
          class="_list-item-text_fcc54c"
          href="https://www.facebook.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          facebook.com
        </a>
      </div>
      <div
        class="_list-item_fcc54c"
      >
        <div
          class="_circle_fcc54c"
        />
        <a
          class="_list-item-text_fcc54c"
          href="https://twitter.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          twitter.com
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`WorkingOn > should render WorkingOn when the dapplet settings is closed 1`] = `
<DocumentFragment>
  <div
    class="_root_fcc54c _ds-closed_fcc54c "
  >
    <span
      class="_title_fcc54c"
    >
      Working on
    </span>
    <div
      class="_list_fcc54c"
    >
      <div
        class="_list-item_fcc54c"
      >
        <div
          class="_circle_fcc54c"
        />
        <a
          class="_list-item-text_fcc54c"
          href="https://telegram.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          telegram.org
        </a>
      </div>
      <div
        class="_list-item_fcc54c"
      >
        <div
          class="_circle_fcc54c"
        />
        <a
          class="_list-item-text_fcc54c"
          href="https://www.whatsapp.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          whatsapp.com
        </a>
      </div>
      <div
        class="_list-item_fcc54c"
      >
        <div
          class="_circle_fcc54c"
        />
        <a
          class="_list-item-text_fcc54c"
          href="https://www.facebook.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          facebook.com
        </a>
      </div>
      <div
        class="_list-item_fcc54c"
      >
        <div
          class="_circle_fcc54c"
        />
        <a
          class="_list-item-text_fcc54c"
          href="https://twitter.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          twitter.com
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
